frappe/utils/data.py

```diff
@@ -2486,18 +2486,21 @@ def is_site_link(link: str) -> bool:
 	return urlparse(link).netloc == urlparse(frappe.utils.get_url()).netloc
 
 
-def add_trackers_to_url(url: str, source: str, campaign: str, medium: str = "email") -> str:
+def add_trackers_to_url(url: str, source: str, campaign: str, medium: str, content: str) -> str:
```

Comment:
Should be optional?

Reply:
Yep, https://github.com/frappe/frappe/pull/24521/commits/0a9383fe2da2c5f42085e1acfe7c5d5e87b89458 fixes the optional implementation. Only source is _actually_ customarily required (e.g. Plausible, GA)

Consistency is provided in https://github.com/frappe/frappe/pull/24521/commits/f421951ee14774c99f36b72db6ab1681d66fc842 by not lowercasing the medium in the js code path, but not in the python code path.

Some real world usage also revealed problems with parameter quoting, notably spaces, addressed in https://github.com/frappe/frappe/pull/24521/commits/15a7b46a1827babb03aa279c72bdc9b71360e7c6. Note that python quotes spaces as `+` while js more stingently as `%20`. But I didn't bother to conform as support for the non-quite-standard `+` is close to ubiquitous.